queue/driver_mongo.go

```diff
@@ -735,9 +739,106 @@ func (d *mongoDriver) Jobs(ctx context.Context) <-chan amboy.Job {
 			"message":   "database interface error",
 		}))
 	}()
+
 	return output
 }
 
+func (d *mongoDriver) getRetryingQuery() bson.M {
+	q := bson.M{
+		"status.completed":       true,
+		"retry_info.retryable":   true,
+		"retry_info.needs_retry": true,
+	}
+	d.modifyQueryForGroup(q)
+	return q
+}
+
+func (d *mongoDriver) RetryableJobs(ctx context.Context, filter RetryableJobFilter) <-chan amboy.RetryableJob {
+	jobs := make(chan amboy.RetryableJob)
+
+	go func() {
```

Comment:
Should this handle panics?

Reply:
I mostly just copy-pasted `Jobs()`, but I can add panic handlers to both of these.